### PR TITLE
 task 4 completed w error handle

### PR DESCRIPTION
**Pull request by oaissnk**

### CODE REVIEW

File: db/app.controller.js [hunk 1]
```diff
@@ -1,21 +1,29 @@
...
       res.status(200).send({ topics });
     })
-    .catch((err) => {
-      next(err);
-    });
+    .catch(next)
 };
 
+exports.getArticles = (req, res, next) => {
+  console.log(req.params, "reqparams")
```
Comment (Hard-Luck) [inline db/app.controller.js]: remove the console logs

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Your articles shouldn't have a body key on them and should also have a comment count on them

File: db/app.test.js [hunk 2]
```diff
@@ -26,6 +25,56 @@ describe("GET /api/topics", () => {
...
+    })  
+})
+describe("GET /api/articles/:article_id", () => {
+    test("200 responds with the object of the article with the path of the article id", () => {
+        return request(app)
+            .get('/api/articles/1')
+            .expect(200)
+            .then(({ body: {articles} }) => {
+                expect(articles).toHaveLength(1)
+                    expect(articles[0]).toEqual({
```
Comment (yalicee) [inline db/app.test.js]: This should not be an array if its articleById

File: db/app.test.js [hunk 3]
```diff
@@ -26,6 +25,56 @@ describe("GET /api/topics", () => {
...
+            .get('/api/articles')
+            .expect(200)
+            .then(({ body: {articles} }) => {
+                expect(articles).toHaveLength(13)
+                    expect(articles[5]).toEqual({
+                        article_id: 6,
+                        title: "A",
+                        topic: "mitch",
+                        author: "icellusedkars",
+                        body: "Delicious tin of cat food",
```
Comment (yalicee) [inline db/app.test.js:40]: There should be no body key and there should be a comment count key

File: db/app.models.js [hunk 4]
```diff
@@ -6,3 +6,17 @@ exports.selectAllTopics = () => {
...
+exports.selectArticles = (id) => {
+let query = "SELECT * FROM articles"
+
+const queryValues = []
+
+if (id){
+  query += " WHERE article_id = $1";
+  queryValues.push(id)
+}
+  query += " ;"
```
Comment (yalicee) [inline db/app.models.js:18]: This if statement is not necessary

File: db/app.controller.js [hunk 5]
```diff
@@ -1,21 +1,28 @@
...
-    .catch((err) => {
-      next(err);
-    });
+    .catch(next)
 };
 
+exports.getArticles = (req, res, next) => {
+  selectArticles(req.params.article_id)
+  .then((articles) => {
+    console.log(articles, "articles")
```
Comment (yalicee) [inline db/app.controller.js]: remove all console.logs

**Review by yalicee — CHANGES_REQUESTED** [review]
- test for 404 id not found (for getArticleById)
- It seems like you haven't done all the changes that were previously requested. Please can you do this before moving on
- I am super confused this is supposed to be getArticles or getArticlebyId. Please can you just do one ticket

File: db/app.models.js (hunk 4 above)
Comment (oaissnk) [reply db/app.models.js:18]: im trying to create a model function for getting all articles and a single one

File: db/app.test.js (hunk 3 above)
Comment (oaissnk) [reply db/app.test.js:40]: in the learn to code it says it should have a body key, can you show me what it should look like please?

File: db/app.models.js (hunk 4 above)
Comment (MKD01) [reply db/app.models.js:18]: These two model functions should not be the same and should be separated, please revisit the learn to code instructions for more detail.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Please do not submit multiple tickets in one pull request. Separate out each ticket and submit them as their own pull requests.

**Pull request merged**